docs/03-onchain-architecture.md

```diff
@@ -0,0 +1,94 @@
+# 3. Smart Contract Architecture
+
+This section describes the Guild's current smart contract architecture. You can learn more about the pilot’s architecture [here](https://protocol-guild.readthedocs.io/en/latest/06-resources.html#pilot).
+
+Protocol Guild uses smart contracts created by [Splits](https://splits.org/) to trustlessly manage the vesting and distribution of donated funds. All donations pass through a [4-year vesting contract](https://app.splits.org/accounts/0x25941dc771bb64514fc8abbce970307fb9d477e9/) on mainnet, which vests funds into a [pass-through wallet](https://app.splits.org/accounts/0x25941dc771bb64514fc8abbce970307fb9d477e9/), which in turn sends funds to a [split contract](https://app.splits.org/accounts/0xd4ad8daba9ee5ef16bb931d1cbe63fb9e102ec10/) for distribution to the membership.
+
+The split contract itself also serves as the Guild’s onchain membership registry of Ethereum’s active L1 R&D maintainers, while a [Moloch v3 DAO](https://admin.daohaus.club/#/molochv3/0x1/0x412a32dd71357bd12337f4408168df903f90cbd3/members) is used to ratify changes to the membership onchain.
+
+## 3.1 Modules
+
+The Guild’s smart contract architecture is modularized as follows:
+
+<img width="1000" alt="OnchainArchitecture" src="https://github.com/cheeky-gorilla/membership/assets/76262359/66cb927d-6e12-4255-9e9f-04f5d6eb676e">
+
+### Vesting Contract
+
+- Contract address: [0x25941dc771bb64514fc8abbce970307fb9d477e9](https://app.splits.org/accounts/0x25941dc771bb64514fc8abbce970307fb9d477e9/)
+- Official documentation: [https://docs.splits.org/core/vesting](https://docs.splits.org/core/vesting)
+
+The Guild’s donation address is an immutable vesting contract which trustlessly vests donated funds over 4 years. Here, trustlessly means donations cannot be stopped or otherwise redirected during the vest by the donor or the recipient once sent.
+
+Anyone can donate ETH and ERC-20 tokens to the vesting contract on mainnet, though NFT donations are not supported - standard NFT transfers (safeTransfer) will be rejected by the contract (i.e. meaning the transaction will fail), while non-safeTransfer NFT donations will be lost.
+
+**How it works:**
+- Donated tokens will accrue in a per-asset queue until a vesting stream is started for that batch of tokens, by triggering the `startStream` function (permissionless, as in any actor can trigger this function, regardless of whether they are a Guild member)
+- Once the vesting stream is started, the tokens will vest linearly over 4 years, and anyone can permissionlessly trigger `releaseStream` to push any vested tokens into a pass-through wallet (this must be done per vesting stream)
+
+All donated tokens are thus forced to vest - there is no way to do anything with them until they are vested into the pass-through wallet.
```

Comment:
We could perhaps remove this given we link to the more exhaustive Splits docs?

Reply:
Keeping for now for consistency, since we have a "how it works" summary for each contract sub-section here, and maybe it's helpful for a high-level overview!